### PR TITLE
chore: Replace Promise.all with Promise.allSettled by using MultiRequ…

### PR DESCRIPTION
**Pull request by Lukasz2891**
…estFetcher

### CODE REVIEW

**Comment by Lukasz2891** [comment]
I'll create a task for implementing unit tests for particular parts of the MultiRequestFetcher and their implementations.

File: packages/oracle-node/src/fetchers/MultiRequestFetcher.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export interface DataProcessor {
+  process(data: any, pricesObj: PricesObj): PricesObj;
+}
+
+export interface RequestFactory {
+  makeRequest(id: string): Promise<any>;
+}
+
+export abstract class MultiRequestFetcher extends BaseFetcher {
+  protected abstract getRequestFactory(): RequestFactory;
```
Comment (hatskier) [inline packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:13]: What is the added value to have the factory and the processor instance here comparing to just having 2 new abstract functions in this class called `makeRequest` and `processResponse`/

Comment (hatskier) [reply packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:13]: I like the idea of this refactoring, but let's please try to not create additional abstractions if they are not needed

File: packages/oracle-node/src/fetchers/MultiRequestFetcher.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  async extractPrices(responses: any): Promise<PricesObj> {
+    const valueProcessor = await this.getDataProcessor();
+    let result: PricesObj = {};
+
+    for (const response of responses) {
+      if (
+        response.status === "rejected" ||
+        response.value === undefined ||
+        response.value.data === undefined
```
Comment (hatskier) [inline packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:35]: Are you sure that each redsponse.value should have the `data` field? It looks a bit axios-specific

File: packages/oracle-node/src/fetchers/MultiRequestFetcher.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { BaseFetcher } from "./BaseFetcher";
```
Comment (hatskier) [inline packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:1]: Just another question. Have you tested this new code version with the main manifest? If not, ca you please doublecheck if everything works as expected before the merge?

**Review by hatskier — CHANGES_REQUESTED** [review]

File: packages/oracle-node/src/fetchers/MultiRequestFetcher.ts (hunk 1 above)
Comment (Lukasz2891) [reply packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:13]: sometimes I need more variables to create the request. The interfaces in a great way hide that. I like my solution :(

File: packages/oracle-node/src/fetchers/MultiRequestFetcher.ts (hunk 3 above)
Comment (Lukasz2891) [reply packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:1]: Why it couldn't work? Could you help me in that?

Comment (Lukasz2891) [reply packages/oracle-node/src/fetchers/MultiRequestFetcher.ts:1]: Now it seems to work properly.

**Review by hatskier — APPROVED** [review]

**Pull request merged**